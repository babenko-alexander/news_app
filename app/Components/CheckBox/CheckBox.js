import React, { PureComponent } from 'react';
import { TouchableOpacity, View, Text, Image } from 'react-native';

export default class CheckBox extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    const checkedIcon = <Image style={{ width: 20, height: 20 }} source={require('../../assets/checked.png')} />;
    const uncheckedIcon = <Image  style={{ width: 20, height: 20 }} source={require('../../assets/unchecked.png')} />;
    return (
      <View style={{ marginBottom: 11, marginRight: 8 }}>
      <TouchableOpacity
        style={{ borderRadius: 50, flexDirection: 'row', alignItems: 'center' }}
        activeOpacity={0.7}
        onPress={this.props.onPress}
      >
        <View>
        {this.props.checked ? checkedIcon : uncheckedIcon}
        </View>
      <View style={{ marginLeft: 5 }}>
        <Text>{this.props.text}</Text>
      </View>
      </TouchableOpacity>
    </View>
    );
  }
}
