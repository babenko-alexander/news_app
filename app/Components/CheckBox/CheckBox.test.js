import React from 'react';
import {shallow} from 'enzyme';
import CheckBox from './CheckBox';
// import { NativeModules } from 'react-native';

describe('CheckBox', () => {
        it('should match to snapshot - Primary', () => {
            const component = shallow(<CheckBox text="test text"/>)
            expect(component).toMatchSnapshot("Primary CheckBox snapshot")
        });
        it('should match to snapshot - Secondary', () => {
            const component = shallow(<CheckBox text="test text" checked={true} />)
            expect(component).toMatchSnapshot("Secondary CheckBox snapshot")
        });
        it('should call onPress', () => {
            // Arrange
            const mockOnPress = jest.fn();     
            const component = shallow(<CheckBox 
                text= "test text" 
                onPress={mockOnPress}          
            />);
            component.instance().props.onPress();
            expect(mockOnPress).toHaveBeenCalled();
        });
        // it('should call saveToPhone', () => {
        //     jest.mock('NativeModules', () => {
        //         return {
        //           PhotoActions: {
        //             saveToPhone: jest.fn(),
        //           },
        //         };
        //       });
        //     const saveToPhoneSpy = jest.spyOn(NativeModules.PhotoActions, 'saveToPhone');
        //     expect(saveToPhoneSpy()).toHaveBeenCalled();
        // });
});