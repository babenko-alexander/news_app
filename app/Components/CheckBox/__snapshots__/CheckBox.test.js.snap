// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should match to snapshot - Primary: Primary CheckBox snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBox
    text="test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.7}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flexDirection": "row",
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/unchecked.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          <Text>
            test text
          </Text>
        </View>
      </TouchableOpacity>,
      "style": Object {
        "marginBottom": 11,
        "marginRight": 8,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.7,
        "children": Array [
          <View>
            <Image
              source={
                Object {
                  "testUri": "../../../app/assets/unchecked.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>,
          <View
            style={
              Object {
                "marginLeft": 5,
              }
            }
          >
            <Text>
              test text
            </Text>
          </View>,
        ],
        "onPress": undefined,
        "style": Object {
          "alignItems": "center",
          "borderRadius": 50,
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={
                Object {
                  "testUri": "../../../app/assets/unchecked.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "testUri": "../../../app/assets/unchecked.png",
              },
              "style": Object {
                "height": 20,
                "width": 20,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text>
              test text
            </Text>,
            "style": Object {
              "marginLeft": 5,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "test text",
            },
            "ref": null,
            "rendered": "test text",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.7}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 50,
              "flexDirection": "row",
            }
          }
        >
          <View>
            <Image
              source={
                Object {
                  "testUri": "../../../app/assets/unchecked.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": 5,
              }
            }
          >
            <Text>
              test text
            </Text>
          </View>
        </TouchableOpacity>,
        "style": Object {
          "marginBottom": 11,
          "marginRight": 8,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.7,
          "children": Array [
            <View>
              <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/unchecked.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />
            </View>,
            <View
              style={
                Object {
                  "marginLeft": 5,
                }
              }
            >
              <Text>
                test text
              </Text>
            </View>,
          ],
          "onPress": undefined,
          "style": Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/unchecked.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": Object {
                  "testUri": "../../../app/assets/unchecked.png",
                },
                "style": Object {
                  "height": 20,
                  "width": 20,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text>
                test text
              </Text>,
              "style": Object {
                "marginLeft": 5,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test text",
              },
              "ref": null,
              "rendered": "test text",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`CheckBox should match to snapshot - Secondary: Secondary CheckBox snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBox
    checked={true}
    text="test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.7}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flexDirection": "row",
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/checked.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          <Text>
            test text
          </Text>
        </View>
      </TouchableOpacity>,
      "style": Object {
        "marginBottom": 11,
        "marginRight": 8,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.7,
        "children": Array [
          <View>
            <Image
              source={
                Object {
                  "testUri": "../../../app/assets/checked.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>,
          <View
            style={
              Object {
                "marginLeft": 5,
              }
            }
          >
            <Text>
              test text
            </Text>
          </View>,
        ],
        "onPress": undefined,
        "style": Object {
          "alignItems": "center",
          "borderRadius": 50,
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={
                Object {
                  "testUri": "../../../app/assets/checked.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "testUri": "../../../app/assets/checked.png",
              },
              "style": Object {
                "height": 20,
                "width": 20,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text>
              test text
            </Text>,
            "style": Object {
              "marginLeft": 5,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "test text",
            },
            "ref": null,
            "rendered": "test text",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.7}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 50,
              "flexDirection": "row",
            }
          }
        >
          <View>
            <Image
              source={
                Object {
                  "testUri": "../../../app/assets/checked.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": 5,
              }
            }
          >
            <Text>
              test text
            </Text>
          </View>
        </TouchableOpacity>,
        "style": Object {
          "marginBottom": 11,
          "marginRight": 8,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.7,
          "children": Array [
            <View>
              <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/checked.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />
            </View>,
            <View
              style={
                Object {
                  "marginLeft": 5,
                }
              }
            >
              <Text>
                test text
              </Text>
            </View>,
          ],
          "onPress": undefined,
          "style": Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/checked.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": Object {
                  "testUri": "../../../app/assets/checked.png",
                },
                "style": Object {
                  "height": 20,
                  "width": 20,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text>
                test text
              </Text>,
              "style": Object {
                "marginLeft": 5,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test text",
              },
              "ref": null,
              "rendered": "test text",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
